src/thor/map_matcher.cc

```diff
@@ -133,15 +134,34 @@ MapMatcher::FormPath(meili::MapMatcher* matcher,
   // Set costing based on the mode
   const auto& costing = mode_costing[static_cast<uint32_t>(mode)];
   bool use_timestamps = options.use_timestamps();
-
   // Iterate through the matched path. Form PathInfo - populate elapsed time
   // Return an empty path (or throw exception) if path is not connected.
   Cost elapsed;
   std::vector<PathInfo> path;
   GraphId prior_edge, prior_node;
-  const NodeInfo* nodeinfo = nullptr;
-  const DirectedEdge* directededge;
   EdgeLabel pred;
+  const GraphTile* tile = nullptr;
+  const DirectedEdge* directededge = nullptr;
+  const NodeInfo* nodeinfo = nullptr;
+
+  // We support either the epoch timestamp that came with the trace point or
+  // a local date time which we convert to epoch by finding the first timezone
+  uint32_t origin_epoch = 0;
+  if (options.shape().begin()->time() != -1.0) {
+    origin_epoch = options.shape().begin()->time();
+  } else if (options.shape().begin()->has_date_time()) {
+    for (const auto& s : edge_segments) {
```

Comment:
here we allow also the date time string as an input for the time of the route. the trick is we need a nodeinfo to get the timezone so we can get the epoch time from it. so we just find the first one and use it. this may be somewhat inaccurate if the beginning of the match has discontinuities but it is less likely that you are flip flopping timezones.